File: tests/fixtures/project_ts/src/auth/logout.ts
/**
 * Logout functionality.
 */

import { decodeToken } from "./jwt";

// In-memory token blacklist (in production, use Redis or similar)
const tokenBlacklist: Set<string> = new Set();

export async function logout(token: string): Promise<void> {
  // Verify the token is valid before blacklisting
  const payload = decodeToken(token);
  if (!payload) {
    throw new Error("Invalid token");
  }
  tokenBlacklist.add(token);
}

export function isTokenBlacklisted(token: string): boolean {
  return tokenBlacklist.has(token);
}

export function clearBlacklist(): void {
  tokenBlacklist.clear();
}

File: tests/fixtures/project_py/src/auth/logout.py
from __future__ import annotations

from src.auth.jwt import decode_token

# In-memory token blacklist (stub — a real app would use Redis or DB)
_blacklisted_tokens: set[str] = set()


def logout(token: str) -> None:
    """Invalidate a token by adding it to the blacklist.

    Raises ValueError if the token is already invalid.
    """
    # Verify the token is valid before blacklisting
    _ = decode_token(token)
    _blacklisted_tokens.add(token)


def is_token_blacklisted(token: str) -> bool:
    """Check whether a token has been blacklisted."""
    return token in _blacklisted_tokens
